arbnode/dataposter/data_poster.go

```diff
@@ -194,13 +194,13 @@ func (p *DataPoster[Meta]) getFeeAndTipCaps(ctx context.Context, lastTipCap *big
 	config := p.config()
 	// MaxFeeCap = (BacklogOfBatches^2 * UrgencyGWei^2 + TargetPriceGWei) * GWei
 	maxFeeCap :=
-		new(big.Int).Mul(
-			new(big.Int).Add(
-				new(big.Int).Mul(
-					new(big.Int).SetUint64(backlogOfBatches*backlogOfBatches),
-					new(big.Int).SetUint64(uint64(config.UrgencyGwei*config.UrgencyGwei))),
-				new(big.Int).SetUint64(uint64(config.TargetPriceGwei))),
-			new(big.Int).SetUint64(uint64(params.GWei)))
+		arbmath.BigMulByFloat(
+			arbmath.BigAddByFloat(
+				arbmath.BigMulByFloat(
+					arbmath.UintToBig(arbmath.SquareUint(backlogOfBatches)),
+					arbmath.SquareFloat(config.UrgencyGwei)),
+				config.TargetPriceGwei),
```

Comment:
Hmm, I didn't realize these methods were truncating earlier. Instead of doing everything in big ints we should probably do everything in floats, multiply by params.Gwei, and then do a saturating cast to a uint64 (if it's bigger than math.MaxUint64 then just use that) and convert it to a big int. That would avoid truncation until it's no longer possible.

Reply:
It's also fine to have a loss of precision due to floats here, as the gas price slightly differing won't matter.